### DISCUSSION TITLE
What does `act` and `actInv` do?

### DISCUSSION THREAD

**Discussion by BolunDai0216** [stack-of-tasks/pinocchio#1901] [Q&A] [answered]
I see a lot of code that has lines like

```python
data.oMf[bl_id].actInv(data.oMf[id].act(l_sp__F))
```

and

```python
data.oMi[joint.id].actInv(data.oMf[bl_id].act(fext__BL))
```

I was trying to find more information about `actInv` and `act` in the documentation, but all I found for `act` is `ay = aXb.act(by)` and for `actInv` is `by = aXb.actInv(ay)`. I am not quite sure what those mean. Is it possible to help me better understand it. Is it trying to say $^ay =\ ^aX^b\texttt{.act(}^by\texttt{)}$ and  $^by =\ ^aX^b\texttt{.actInv(}^ay\texttt{)}$ where $^ay = {^aX^b}{^by}$?

Thanks in advance!

**Comment by MegMll** [comment, marked as answer]
Hi, 

I think you understood correctly what those functions do.

Using aXb.act() will transform coordinates of point p in frame B into coordinates of point p in frame A. 
${}^ay = {}^aM_b{}^bp$

And aXb.actInv() will transform coordinates of point p in frame A into coordinates of point p in frame B.
${}^by = {}^bM_a{}^ap$

**Comment by fabinsch** [comment]
If you are not sure about the usage of a concept like `act` in your case, you find most of the time some explanations in the python docstring.
```
In [1]: import pinocchio as pin

In [2]: pin.SE3.act?
Docstring:
act( (SE3)self, (numpy.ndarray)point) -> numpy.ndarray :
    Returns a point which is the result of the entry point transforms by *this.

act( (SE3)self, (SE3)M) -> SE3 :
    Returns the result of *this * M.

act( (SE3)self, (Motion)motion) -> Motion :
    Returns the result action of *this onto a Motion.

act( (SE3)self, (Force)force) -> Force :
    Returns the result of *this onto a Force.

act( (SE3)self, (Inertia)inertia) -> Inertia :
    Returns the result of *this onto a Force.
Type:      function
```
I hope this can additionally help you.
